packages/devkit/src/utils/package-json.ts

```diff
@@ -314,27 +314,20 @@ export async function ensurePackage(
     throwOnMissing?: boolean;
   } = {}
 ): Promise<void> {
-  let version: string;
+  let version: string = null;
 
   // Read package and version from root package.json file.
-  const packageJson = readJson(tree, 'package.json');
   const dev = options.dev ?? true;
   const throwOnMissing = options.throwOnMissing ?? !!process.env.NX_DRY_RUN; // NX_DRY_RUN is set in `packages/nx/src/command-line/generate.ts`
   const pmc = getPackageManagerCommand();
-  const field = dev ? 'devDependencies' : 'dependencies';
 
-  version = packageJson[field]?.[pkg];
-
-  // If package not found, try to resolve it using Node and get its version.
-  if (!version) {
-    try {
-      version = require(`${pkg}/package.json`).version;
-    } catch {
-      // ignore
-    }
+  try {
+    version = require(`${pkg}/package.json`).version;
```

Comment:
Use readModulePackageJson from `packages/nx/src/utils/package-json.ts` this will throw for some packages.